tests/Metadata/Extractor/ResourceMetadataCompatibilityTest.php

```diff
@@ -654,4 +663,23 @@ private function withGraphQlOperations(array $values, ?array $fixtures): array
 
         return $operations;
     }
+
+    private function withPersistenceMeans(array $values): ?PersistenceMeansInterface
+    {
+        if (!$values) {
+            return null;
+        }
+
+        if (1 !== \count($values)) {
+            throw new \InvalidArgumentException('Only one persistence means can be configured at a time.');
+        }
+
+        $configuration = reset($values);
+        switch (key($values)) {
+            case 'elasticsearchDocument':
+                return new ElasticsearchDocument($configuration['index'] ?? null, $configuration['type'] ?? null);
+        }
+
+        throw new \DomainException(sprintf('Unsupported "%s" persistence means.', key($values)));
```

Comment:
We don't use `DomainException`, `RuntimeException` instead?

Reply:
`LogicException` is better IMHO "This kind of exception should lead directly to a fix in your code."

Reply:
`DomainException`: “Exception thrown if a value does not adhere to a defined valid data domain.”

This is the case here (and it extends `LogicException`).

Reply:
We don't use it in API Platform and I'm Symfony. We reserve it for the app domain.

Reply:
We don't use `DomainException` in the API Platform code base (its definition is not very clear, see  https://stackoverflow.com/a/1103021/1352334 and the related PHP bug). I suggest keeping `LogicException` for consistency.